lib/inc/internal/facts/resolvers/fips_resolver.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+* @file
+* Declares the base fips fact resolver.
+*/
+#pragma once
+
+#include <facter/facts/resolver.hpp>
+
+namespace facter { namespace facts { namespace resolvers {
+
+
+    /**
+     * Responsible for resolving fips facts.
+     */
+    struct fips_resolver : resolver
+    {
+        fips_resolver();
+
+     protected:
+        /**
+         *  Represents fips data.
+         */
+        struct data
+        {
+            /**
+             * Stores the is_fips_mode_enabled data.
+             */
+            bool is_fips_mode_enabled;
+        };
+
+        /**
+         * Called to resolve all facts the resolver is responsible for.
+         * @param facts The fact collection that is resolving facts.
+         */
+        virtual void resolve(collection& facts) override;
```

Comment:
This does not need to be a protected member -- can make it public. See e.g. https://github.com/puppetlabs/facter/blob/master/lib/inc/internal/facts/resolvers/processor_resolver.hpp#L28

Reply:
yikes.